packages/auth/graphql/schema.js

```diff
@@ -41,5 +52,20 @@ type mutations {
   # if userId is null, the logged in user's sessions get cleared
   # required role to clear other's session: supporter
   clearSessions(userId: ID): Boolean!
+
+  # Generate a new sharedSecret for TOTP
+  initTOTPSharedSecret: SharedSecretResponse!
+
+  # Validate the sharedSecret for the first time via token payload
+  validateTOTPSharedSecret(totp: String): Boolean!
+
+  # Activate or deactivate 2FA
+  updateTwoFactorAuthentication(enabled: Boolean!): Boolean!
+
+  # Resend Verification Code to confirm a phone number
+  resendPhoneNumberVerificationCode: Boolean!
```

Comment:
why is this called re-send and not just send?

Reply:
because the initial "send" is usually triggered from the "updatePhoneNumber"

Reply:
Have you considered that the phone number can also be updated with `updateMe`? I'd prefer to keep that as is, drop `updatePhoneNumber` and rename this to `sendPhoneNumberVerificationCode`. Sending on `updatePhoneNumber` feels like a unwanted side-effect. I'm fine with calling `updateMe` and `sendPhoneNumberVerificationCode` in succession in the frontend. We'll need to handle already entered phone number anyway.